performance-tests/bench/report_parser/argument_parser.cpp

```diff
@@ -0,0 +1,136 @@
+#include "argument_parser.h"
+#include <util.h>
+#include <json_conversion.h>
+#include <BenchTypeSupportImpl.h>
+#include <unordered_map>
+
+using namespace Bench;
+
+bool ArgumentParser::parse(int argc, ACE_TCHAR* argv[], OutputType& output_type,
+  OutputFormat& output_format, Report& report, std::ofstream& output_file_stream) {
+  std::string input_file_path;
+  std::string output_file_path;
+
+  if (argc == 1) {
+    show_usage_prompt();
+  } else {
+    try {
+      for (int i = 1; i < argc; i++) {
+        const ACE_TCHAR* argument = argv[i];
+        std::string option_argument = get_option_argument(i, argc, argv);
+
+        if (!ACE_OS::strcmp(argument, ACE_TEXT("--help"))
+          || !ACE_OS::strcmp(argument, ACE_TEXT("-h"))) {
+          std::cout << usage << std::endl
+            << std::endl
+            << "OPTIONS:" << std::endl
+            << "--input-file <filename>      The report file to read" << std::endl
+            << "--output-file <filename>     The parsed file to generate" << std::endl
+            << "--output-type <type>         Specifies type of data to parse." << std::endl
+            << "     Options:" << std::endl
+            << "            time-series:     Parses out time-series data" << std::endl
+            << "--output-format <format>     Specifies format of output." << std::endl
+            << "     Options:" << std::endl
+            << "            gnuplot:         Formats output for plotting with gnuplot" << std::endl;
+          return false;
+        } else if (!ACE_OS::strcmp(argument, ACE_TEXT("--input-file"))) {
+          input_file_path = option_argument;
+        } else if (!ACE_OS::strcmp(argument, ACE_TEXT("--output-file"))) {
+          output_file_path = option_argument;
+        } else if (!ACE_OS::strcmp(argument, ACE_TEXT("--output-type"))) {
+          static std::unordered_map<std::string, OutputType> const table = {
+            { "time-series", OutputType::TimeSeries }
+          };
+
+          auto it = table.find(option_argument);
+
+          if (it != table.end()) {
+            output_type = it->second;
+          } else {
+            show_option_argument_error(option_argument);
+            return false;
+          }
+        } else if (!ACE_OS::strcmp(argument, ACE_TEXT("--output-format"))) {
+          static std::unordered_map<std::string, OutputFormat> const table = {
+            { "gnuplot", OutputFormat::Gnuplot }
+          };
+
+          auto it = table.find(option_argument);
+
+          if (it != table.end()) {
+            output_format = it->second;
+          } else {
+            show_option_argument_error(option_argument);
+            return false;
+          }
+        } else {
+          show_option_error(argument);
```

Comment:
wchar compile issue... easiest fix is probably to wrap argument with an ACE_TEXT_ALWAYS_CHAR()

Reply:
Done.